### ISSUE TITLE
Debugger: DISK ADDR/DATA PROLOGUE/EPILOGUE

### ISSUE THREAD

**Issue by Michaelangel007** [AppleWin/AppleWin#1221]
**TL:DR;** Displaying the Track/Sector number from the VTSC address field (VolumeTrackSectorChecksum) only works for standard disks. Move disk address / data fields to debugger.

While working on #1215 and looking over `DecodeLatchNibble()` in _DiskFormatTrack.cpp I saw that it had [hard-coded](https://github.com/AppleWin/AppleWin/blob/master/source/DiskFormatTrack.cpp#L246-L318) the disk address / data fields. `kADDR_PROLOGUE_DOS3_3`, `0xDEAAEB`, and `0xD5AAAD`.  

I propose that we have 4 new debugger commands to set these:

    DISK ADDR PROLOGUE [bytes] [length]
    DISK ADDR EPILOGUE [bytes] [length]
    DISK DATA PROLOGUE [bytes] [length]
    DISK DATA EPILOGUE [bytes] [length]

With zero args this would reset to the default:

    DISK ADDR PROLOGUE D5AA96 3
    DISK ADDR EPILOGUE DEAAEB 3
    DISK DATA PROLOGUE D5AAAD 3
    DISK DATA EPILOGUE DEAAEB 3

Similar to how Copy ][+'s Sector Editors lets you "Patch DOS" (`P`) to use custom values:

![copy_2_plus_84_sector_editor_patch](https://user-images.githubusercontent.com/7876796/235357689-fc8cd615-19d4-4b39-aa7b-7ef5a1885724.png)

This will let us display the correct Track/Sector for .WOZ disks that use custom formats and when I finally get around to adding disk reading/writing (RWTS) to the debugger this will the debugger read/write those correctly.

    //     DISK # READ  <Track> <Sector> <NumSec> <Addr>     // Read Track/Sector(s)
    //     DISK # READ  <Track> <Sector> Addr:Addr           // Read Track/Sector(s)
    //     DISK # WRITE <Track> <Sector> Addr:Addr           // Write Track/Sector(s)

**Comment by tomcw** [comment]
btw, that `FormatTrack` class was originally written purely to support DOS 3.2/3.3 INIT or ProDOS Format for .DSK and .NIB images. WOZ images don't need such a workaround.

Overloading it's functionality to read the current sector number wasn't in the original design (although I have been using this class in debug/logging mode to output log messages about the current VTSC just read etc).

The bottom line is that it's used for INIT/Formatting .DSK/NIB images, so for any changes made, then please check against the regression tests over in AppleWin-Test:

https://github.com/AppleWin/AppleWin-Test/tree/master/v2/DiskFormat
https://github.com/AppleWin/AppleWin-Test/blob/master/v2/DiskFormat/test.bat

**Comment by Michaelangel007** [comment]
Thanks for info the and link Tom.

I'll play around with AppleWin-Test next weekend.

**Comment by tomcw** [comment]
btw, it's easy to just run a subset of the tests, eg. just the DiskFormat tests...

- checkout the AppleWin-Test repo
- `CD` to the "v2/DiskFormat" folder
- `test.bat <path>`  ; this is the path to your AppleWin.exe you want to test
